README.md

```diff
@@ -557,12 +591,262 @@ end
 
 ---------------------------------------------
 
-![Interactive Tile 2 Output](https://github.com/alinchis/Escher.jl/blob/master/assets/img/interactive_tile_2.png "Interactive Tile 2")
+![Interactive Tile 2 Output](./assets/img/interactive_ui.png "Interactive Tile 2")
 
 ---------------------------------------------
 
 ##### 4.2.5.3. Interactive Tiles
 
 Escher provides ready-to-use `Tiles` like buttons, input, codemirror etc., in the [Widgets API](http://escher-jl.org/widgets-api.html). Keep in mind that, regular and interactive, `Tiles` can be augmented with clickable, selectable, keypress etc. behavior via the [Behavior API](http://escher-jl.org/behavior-api.html).
 
+
 ### 4.3. Escher Advanced
+
+This section is intended to provide a more detailed view on the way Escher works and how to create your own custom assets.
+
+#### 4.3.1. Assets basics
+
+I can think of three reasons why one might need to create new assets, for fun not included:
+
+- if the current APIs do not provide the elements you require, you can build your own.
+- Escher may not give access to all available properties defined in the HTML element, or it may have certain default values that you would like to change.
+- you might want to optimize your code. When loading assets, all templates in the assets package get loaded into the webpage, even the ones that are not used. For the development phase this is not an issue, but in production you might want to provide a faster loading time by creating your package only with the templates you are using.
+
+To create a new asset element you need three things:
+
+1. a custom `HTML` element template you wish to use
+2. a `Tile` type to match the HTML element kind, depending on what kind of interaction is needed. The `Tile` fields do not necessarily mirror the properties of the HTML.
+3. a `render(Tile, state) = Elem(HTML element)` function that does the wiring. The user chooses which `Tile` fields correspond to which `HTML` property.
```

Comment:
this is till complex to understand. You can just say,  "a `render(::MyTile, state)` which returns an `Elem` (link to https://github.com/shashi/Patchwork.jl#creating-elements). "This may create the custom element you created in step 1"?